packages/zowe-explorer/src/configuration/Profiles.ts

```diff
@@ -821,10 +821,11 @@ export class Profiles extends ProfilesCache {
         const profInfo = await this.getProfileInfo();
         const configApi = profInfo.getTeamConfig();
         const profAttrs = await this.getProfileFromConfig(profileName);
-        if (loginTokenType && loginTokenType.startsWith("apimlAuthenticationToken")) {
-            configApi.set(`${profAttrs.profLoc.jsonLoc}.secure`, []);
-        } else {
-            configApi.set(`${profAttrs.profLoc.jsonLoc}.secure`, ["tokenValue"]);
+        if (profAttrs.profLoc.jsonLoc) {
```

Comment:
Is it possible for `profAttrs` to be undefined here?
I guess I'm wondering about it since we did `?.` in `tokenAuthClearSecureArray` 😅

Reply:
It shouldn't be possible as long as a profile matching `profileName` exists - and since we call the function with `serviceProfile.name` as the profile name parameter, ZE should always find attributes for that profile.

See `Profiles.handleSwitchAuthentication`, line 877:

```typescript
const serviceProfile = node.getProfile() ?? this.loadNamedProfile(node.label.toString().trim());
```